### PR TITLE
Fix ci and get working with Ember 5

### PR DESCRIPTION
**Pull request by Gaurav0**
Merges https://github.com/ember-fastboot/ember-cli-fastboot/pull/898 , https://github.com/ember-fastboot/ember-cli-fastboot/pull/907, and https://github.com/ember-fastboot/ember-cli-fastboot/pull/915

fastboot-shoebox-test was broken by https://github.com/ember-fastboot/ember-cli-fastboot/pull/905

Only one test is failing, and it is due to this bug: https://github.com/emberjs/ember.js/issues/20377

Update: the test is "Fixed", but only because the current release is now Ember 5.0

### CODE REVIEW

File: packages/ember-cli-fastboot/test/fastboot-location-config-test.js [hunk 1]
```diff
@@ -16,15 +16,15 @@ describe('FastBootLocation Configuration', function () {
 
     return app
       .create('fastboot-location-config', {
-        emberVersion: 'latest',
-        emberDataVersion: 'latest',
+        emberVersion: '~3.28.12',
```
Comment (kiwiupover) [inline packages/ember-cli-fastboot/test/fastboot-location-config-test.js:19]: @Gaurav0, thanks heaps for doing this. 

Why fix the version to 3.?

**Review by kiwiupover — APPROVED** [review]
Thanks heaps for doing this @Gaurav0 Looks great to me

File: packages/ember-cli-fastboot/test/fastboot-location-config-test.js (hunk 1 above)
Comment (Gaurav0) [reply packages/ember-cli-fastboot/test/fastboot-location-config-test.js:19]: It matches the default version of the ember-cli-fastboot addon being used in the test.

In a followup PR, we should probably update all the defaults to 4.12. Right now, just want to get things working again.

**Comment by gilest** [comment]
Nice work @Gaurav0 !

How can we help with this? I'm happy to test it with our app if that's helpful.

**Comment by mansona** [comment]
I'm closing this for now. I've looked into it and it seems like the fix is in the wrong place. Also the PR is too big to realistically review properly.

For anyone following this work I'm taking back up the mantle of fixing ember-cli-fastboot this week so don't worry about this getting lost 👍

**Pull request closed without merging**